param_passing/proposal.md

```diff
@@ -0,0 +1,194 @@
+        <plugin name="camera_depth_sensor" filename="libcamera_depth_sensor.so"/>
+     </sensor>
+
+      <visual name="chassis::camera_visual" action="add">
+        <pose>0.5 0.02 0 0 0</pose>
+        <geometry>
+            <box>
+                <size>0.02 0.02 0.02</size>
+            </box>
+        </geometry>
+      </visual>
+
+      <sensor name="top::camera" action="modify">
+        <pose>0 1 0 0 0 0</pose>
+        <update_rate>60</update_rate>
+      </sensor>
+
+      <visual name="top::camera_visual">
+        <geometry action="replace">
+          <sphere>
+            <radius>0.05</radius>
+          </sphere>
+        </geometry>
+        <material action="modify">
+          <ambient>0.0 1.0 0.0 1</ambient>
+        </material>
+      </visual>
+
+    </experimental:params>
+  </include>
+</model>
+```
+
+In `//include/experimental::params`, the action can be either listed in the element identifier or listed individually in the children element(s). For example,
+
+```xml
+<sensor name="top::camera" action="modify">
+  <pose>0 1 0 0 0 0</pose>
+  <update_rate>60</update_rate>
+</sensor>
+```
+
+The action is defined at the top level where the `name` identifier is, which means that all children listed in this element will be modified. If the user wants to do several different actions to the children elements then the user can specify the action for each child element individually. For instance,
+
+```xml
+<visual name="top::camera_visual">
+  <geometry action="replace">
+    <sphere>
+      <radius>0.05</radius>
+    </sphere>
+  </geometry>
+  <material action="modify">
+    <ambient>0.0 1.0 0.0 1</ambient>
+  </material>
+</visual>
+```
+
+If the original model has several corresponding elements, it will be up to the user to update all appropriately. For example, if a lidar has a `//visual` and `//sensor` element and the user would like to remove the lidar information, then both will need to be listed for removal:
+
+```xml
+<visual name="chassis::lidar_visual" action="remove"/>
+<sensor name="chassis::lidar" action="remove"/>
+```
+
+When a user wants to `add` an element that does not exist in the original model, then the `add` action can be used similarly as follows:
+
+```xml
+<visual name="chassis::camera_visual" action="add">
+  <pose>0.5 0.02 0 0 0</pose>
+  <geometry>
+      <box>
+          <size>0.02 0.02 0.02</size>
+      </box>
+  </geometry>
+</visual>
+```
+
+In this example, the `//visual` element named "camera_visual" will be added as a child of `//link[@name=’chassis’]`. A check will be performed to ensure that the element does not already exist. If it does exist, then a warning/error will be printed and the element will be skipped. The process of skipping elements will also occur when the user uses the `modify`, `replace`, and/or `remove` actions and the element is not found in the original model.
+
+Another example could be adding a plugin to a `//sensor` (which can not be done through model composition):
```

Comment:
This seems a little out of place since adding a plugin to a sensor is already shown in a previous example. I understand this is to show that simple model composition cannot be used to achieve this, so I get why it's here. Maybe it can be in its own subsection with a little more explanation and emphasis on why this example is here?

Reply:
104440aa604acc3814258e6215c2bf5b351ed347